pkg/datapath/linux/config/config.go

```diff
@@ -390,6 +391,17 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 		cDefinesMap["NATIVE_DEV_MAC_BY_IFINDEX(IFINDEX)"] = macByIfIndexMacro
 		cDefinesMap["IS_L3_DEV(ifindex)"] = isL3DevMacro
 	}
+
+	// Enabling NetfilterCompatibleMode increases BPF instruction memory size, and may
+	// cause issues in kernels have lower memory. Hence enabling, this feature
```

Comment:
Not sure what you mean by "kernels have lower memory". The large insn limit is only relevant in the context of the BPF verifier complexity, but not of the memory consumption.

Reply:
I was trying to make a point if we enable this feature on lower version of kernel, BPF code memory consumption increases relatively higher. [unfortunately code memory increased by 1200 bytes]. Though bpf verifier catches this Issue, if user enable configs that match with MAX_HOST_OPTIONS mentioned [here ](https://github.com/cilium/cilium/blob/master/bpf/Makefile#L168) wouldn't cause any Issues host interface ?

Reply:
I think it's the "memory" wording that's confusing, this is also the first time I've heard about any sort of memory concern.

I _suspect_ that what you're referring to is the instruction complexity limit, which is in some sense reflective of the total number of instructions that the kernel must iterate to validate all of the paths through the programs. If this isn't what you mean, I think we'd both be curious to see more detailed output of what you're referring to and where you got the 1200B reference from.

Reply:
I have coined bpf `instruction complexity limit` as bpf `kernel lower memory'. That opens up confusion here.
I'll update the comments in to code to bring more clarity.

Reply:
will fix in separate PR.